lib/WUI/link_content/usb_files.cpp

```diff
@@ -0,0 +1,53 @@
+#include "usb_files.h"
+#include "../nhttp/headers.h"
+
+// Why does FILE_PATH_BUFFER_LEN lives in *gui*!?
+#include <../../src/gui/file_list_defs.h>
+
+namespace nhttp::link_content {
+
+using std::nullopt;
+using std::optional;
+using std::string_view;
+using namespace handler;
+
+optional<ConnectionState> UsbFiles::accept(const RequestParser &parser) const {
+    const string_view uri = parser.uri();
+    const constexpr string_view prefix = "/usb/";
+
+    // We claim the /usb/ namespace.
+    if (uri.size() < prefix.size() || uri.substr(0, prefix.size()) != prefix) {
+        return nullopt;
+    }
+
+    // Content of the USB drive is only for authenticated, don't ever try anything without it.
```

Comment:
Should we move the authentication to the top of the function? Before checking if the path is correct.

Reply:
Reading it again (and changing it and running tests), no.

The thing above isn't checking for correctness (if the file exists). It checks if the URL starts with `/usb`. If not (eg. it's `/api/` or `/index.html`), it doesn't „belong“ to this selector and it can't really say anything about it -> it returns `nullopt` and „refuses“ the URL.